Core/Quest/Objects/Creatures Forge Camp Annihilated

poszer opened the pull request:
Fixed Quest Forge Camp Annihilated: The Canons now shoot the camps

<!-- First of all, THANK YOU for your contribution.
 Please fill this template unless your PR is very simple/straightforward.
 Do not forget to have a look at our Pull Request tutorial: https://github.com/azerothcore/azerothcore-wotlk/wiki/Contribute#how-to-create-a-pull-request
-->


<!-- WRITE A RELEVANT TITLE -->


##### CHANGES PROPOSED:

- Updated SAI Script
- Updated Creatures ( Problem with LOS Targeting ) - Moved to Clear positions where can be targeted.
- Creature 68741  ( Fel Cannon - Fear Target )
- Creature 68742  ( Fel Cannon - Hate Target )

###### ISSUES ADDRESSED:
<!-- If the issue doesn't exist, describe it and how to reproduce it, please. If the issue already exists, just paste the link to the issue you close, like this: Closes https://github.com/azerothcore/azerothcore-wotlk/issues/967 -->

Closes 
https://github.com/azerothcore/azerothcore-wotlk/issues/1077

##### TESTS PERFORMED:
<!-- Does it build without errors? Did you test in-game? What did you test? Did you do all these tests on Linux, Mac or Windows? Other tests performed -->
Builded without Errors, tested on two cores, everything is fine and working just as like blizzlike.
The Quest can be done without problem & Cannons now shoot the camps.

##### HOW TO TEST THE CHANGES:
<!-- We need to confirm the changes first, so try to make the work easy for testers, please:
 - Which commands to use? Which NPC to teleport to?
 - Do we need to enable debug flags on Cmake?
 - Do we need to look at the console? etc...
 - Other steps
-->
1. .quest add 10011
2. .additem 25770
3. .go creature 68367
4. Use the Item from your backpack ( Fel Cannon Activator )
5. .additem 25771
6. .go creature 68740
7. Use the Item from your backpack ( Use Fel Cannon Activator )
8. The both should work and shoot the camps


##### KNOWN ISSUES AND TODO LIST:
<!-- This is a TODO list with checkboxes to tick -->

- [ ]
- [ ] 


##### Target branch(es):

Master


<!-- NOTE: You no longer need to squash your commits, on merge we will squash them for you (when there are too many commits we merge them into one big commit for a cleaner and easy to read history). -->

<!-- NOTE2: If you intend to contribute more than once, you should really join us on our discord channel!
 The link is on our site http://azerothcore.org/ We set cosmetic ranks for our contributors and may give access to special resources/knowledge to them! -->

Stoabrogga commented:
First of all please change the comment field of the statements, they are mentioning Fear for both smart scripts. I also had to remove the single quotes for the position values in the last two statements, as these are numeric values.
After applying the statements the cannons shoot, but the beam is too high and the target is not bursting in flames as it is shown in this video: https://youtu.be/LG2_vxFTleo?t=24

poszer commented:
> 
> 
> First of all please change the comment field of the statements, they are mentioning Fear for both smart scripts. I also had to remove the single quotes for the position values in the last two statements, as these are numeric values.
> After applying the statements the cannons shoot, but the beam is too high and the target is not bursting in flames as it is shown in this video: https://youtu.be/LG2_vxFTleo?t=24

Oh, thanks. Edited

Stoabrogga commented:
The script still needs the following improvements:
- Adjust the beam to have it's start point within the cannon
- Engulf the targeted camps in flames after hit by the cannon

I could also take a look at this if I find the time.

poszer commented:
> 
> 
> Please use this SQL script instead:
> 
> ```
> INSERT INTO version_db_world (`sql_rev`) VALUES ('1545767614007203800');
> 
> DELETE FROM `smart_scripts` WHERE `entryorguid` = 19210;
> INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`)
> VALUES
> (19210, 0, 0, 1, 8, 0, 100, 0, 33532, 0, 0, 0, 33, 19210, 0, 0, 0, 0, 0, 16, 0, 0, 0, 0, 0, 0, 0, 'On spell \'Fel Cannon: Fear\' (33532) hit  - Party invoker: Give kill credit \'Fel Cannon: Fear\' (19210)'),
> (19210, 0, 1, 2, 61, 0, 100, 0, 0, 0, 0, 0, 66, 0, 0, 0, 0, 0, 0, 19, 19211, 100, 0, 0, 0, 0, 0, 'Linked - Self: Look at closest alive creature \'Fel Cannon: Fear Target\' (19211) in 100 yards'),
> (19210, 0, 2, 0, 61, 0, 100, 0, 0, 0, 0, 0, 22, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'Linked - Self: Set event phase to 2'),
> (19210, 0, 3, 0, 1, 2, 100, 0, 3000, 3000, 10000, 10000, 11, 33535, 0, 0, 0, 0, 0, 19, 19211, 100, 0, 0, 0, 0, 0, 'When out of combat and timer at the begining between 3000 and 3000 ms (and later repeats every 10000 and 10000 ms) - Self: Cast spell \'Fel Energy Beam\' (33535) on closest alive creature \'Fel Cannon: Fear Target\' (19211) in 100 yards'),
> (19210, 0, 4, 5, 1, 2, 100, 0, 8000, 8000, 10000, 10000, 92, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'When out of combat and timer at the begining between 8000 and 8000 ms (and later repeats every 10000 and 10000 ms) - Self: Interrupt cast spell'),
> (19210, 0, 5, 6, 61, 2, 100, 0, 0, 0, 0, 0, 86, 42346, 0, 19, 19211, 100, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Linked - Cross Cast \'Cosmetic - Flame Patch 2.0\' (42346) on closest alive creature \'Fel Cannon: Fear Target\' (19211) in 100 yards'),
> (19210, 0, 6, 7, 61, 0, 100, 0, 0, 0, 0, 0, 22, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'Linked - Self: Set event phase to 1'),
> (19210, 0, 7, 0, 61, 0, 100, 0, 0, 0, 0, 0, 66, 0, 0, 0, 0, 0, 0, 8, 0, 0, 0, 0, 0, 0, 1.09956, 'Linked - Self: Reset to original orientation 1.09956');
> 
> DELETE FROM `smart_scripts` WHERE `entryorguid` = 19067;
> INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`)
> VALUES
> (19067, 0, 0, 1, 8, 0, 100, 0, 33531, 0, 0, 0, 33, 19067, 0, 0, 0, 0, 0, 16, 0, 0, 0, 0, 0, 0, 0, 'On spell \'Fel Cannon: Hate\' (33531) hit  - Party invoker: Give kill credit \'Fel Cannon: Hate\' (19067)'),
> (19067, 0, 1, 2, 61, 0, 100, 0, 0, 0, 0, 0, 66, 0, 0, 0, 0, 0, 0, 19, 19212, 100, 0, 0, 0, 0, 0, 'Linked - Self: Look at closest alive creature \'Fel Cannon: Hate Target\' (19212) in 100 yards'),
> (19067, 0, 2, 0, 61, 0, 100, 0, 0, 0, 0, 0, 22, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'Linked - Self: Set event phase to 2'),
> (19067, 0, 3, 0, 1, 2, 100, 0, 3000, 3000, 10000, 10000, 11, 33535, 0, 0, 0, 0, 0, 19, 19212, 100, 0, 0, 0, 0, 0, 'When out of combat and timer at the begining between 3000 and 3000 ms (and later repeats every 10000 and 10000 ms) - Self: Cast spell \'Fel Energy Beam\' (33535) on closest alive creature \'Fel Cannon: Hate Target\' (19212) in 100 yards'),
> (19067, 0, 4, 5, 1, 2, 100, 0, 8000, 8000, 10000, 10000, 92, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'When out of combat and timer at the begining between 8000 and 8000 ms (and later repeats every 10000 and 10000 ms) - Self: Interrupt cast spell'),
> (19067, 0, 5, 6, 61, 2, 100, 0, 0, 0, 0, 0, 86, 42346, 0, 19, 19212, 100, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Linked - Cross Cast \'Cosmetic - Flame Patch 2.0\' (42346) on closest alive creature \'Fel Cannon: Hate Target\' (19212) in 100 yards'),
> (19067, 0, 6, 7, 61, 0, 100, 0, 0, 0, 0, 0, 22, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'Linked - Self: Set event phase to 1'),
> (19067, 0, 7, 0, 61, 0, 100, 0, 0, 0, 0, 0, 66, 0, 0, 0, 0, 0, 0, 8, 0, 0, 0, 0, 0, 0, 0.610865, 'Linked - Self: Reset to original orientation 0.610865');
> ```
> 
> Some additional notes:
> 
>     * It is not necessary to change the position of the beam targets, it suffices to increase the number of yards in the smart script, so I removed the statements.
> 
>     * Sadly, I was not able to adjust the beam. But this seems to have also been an issue in the original version, as seen in this video:
>       https://www.youtube.com/watch?v=jJG9cuhHIgE&t=72
> 
>     * I did not manage to create an equal flame effect as shown in the video, but I think I've found a good alternative.
> 
>     * As an additional entry I added an action to return the fel cannons to their original orientation.
> 
> 
> The SQL script was successfully tested on my own test-server.

Updated, thanks

poszer commented:
Since we don't have better fix for it where we can adjust the beam, this should be okay. Can be merged @BarbzYHOOL
